src/code/qualification/plotQualificationPKRatio.m

```diff
@@ -77,72 +107,62 @@
         throw(ME);
     end
     
-    % Get the observation time and PK profile in the right unit for study
-    ObsTime = ObservedData.Time(ObservedData.ID==PKRatio.ObservedDataRecordId);
-    ObsTimeUnit = ObservedData.TimeUnit(ObservedData.ID==PKRatio.ObservedDataRecordId);
-    ObsTimeUnit = ObsTimeUnit(1);
-    Obs = ObservedData.Avg(ObservedData.ID==PKRatio.ObservedDataRecordId);
-    if iscell(Obs)
-        Obs = str2double(Obs);
-    end
-    ObsUnit = ObservedData.AvgUnit(ObservedData.ID==PKRatio.ObservedDataRecordId);
-    ObsUnit = ObsUnit{1};
-    
-    % Get comparable units to pred
-    Timefactor=getUnitFactor(ObsTimeUnit,SimTimeUnit,'time');
-    % Because of encoding issues, some time ? is read as ? or ?
-    try
-        Obsfactor=getUnitFactor(ObsUnit,PredUnit,PredDimension, 'MW', MW);
-    catch
-        ObsUnit = convertPKSimUnit(ObsUnit);
-        Obsfactor=getUnitFactor(ObsUnit,PredUnit,PredDimension, 'MW', MW);
-    end
-    ObsTime = ObsTime.*Timefactor;
-    Obs = Obs.*Obsfactor;
-    
     % Get PK parameters from the curves
     % For simulation
     allPKpred=getPKParametersForConcentration(SimTime, Pred, 'Dose', drugmass);
     
-    % For Observations
-    allPKobs=getPKParametersForConcentration(ObsTime, Obs, 'Dose', drugmass);
-    
     for k=1:length(PKParameter)
-        % Get the PK parameters requested in PKParameter
-        if strcmpi(PKParameter{k}, 'AUC')
-            PKpredField{k}= 'AUC_last';
-        elseif strcmpi(PKParameter{k}, 'CMAX')
-            PKpredField{k}= 'cMax';
-        else
-            PKpredField{k}=PKParameter{k};
-        end
-        if isfield(allPKpred, PKpredField{k}) && isfield(allPKobs, PKpredField{k})
+        % Get the PK parameters requested in PK Parameters
+        % according to Obs Unit
+        if strcmp(Result.obsPKDimension(i, k), 'AUC (mass)')
+            AUCpred = getfield(allPKpred, 'AUC_last');
+            AUCpredUnitFactor = getUnitFactor('mg*h/l', Result.obsPKUnit{i, k}, 'AUC (mass)', 'MW', MW);
+            Result.predPK(i, k) = AUCpred.*AUCpredUnitFactor;
+            
+        elseif strcmp(Result.obsPKDimension(i, k), 'Concentration')
+            
+            CMAXpred = getfield(allPKpred, 'cMax');
+            CMAXpredUnitFactor = getUnitFactor('mg/l', Result.obsPKUnit{i, k}, 'Concentration', 'MW', MW);
+            Result.predPK(i, k) = CMAXpred.*CMAXpredUnitFactor;

```

Comment:
I don't understand this change. Please explain.

Reply:
I assumed that Cmax can be a parameter requested in the PK ratio since it can for DDI Ratio.
If Cmax is requested, I convert it into the Observed Unit. And because I wanted the unit as mg, l and h for AUC and CL, the current unit for Cmax would be in mg/l